### PR TITLE
[CDAP-12969] Have a generic popover based on design instead of using reactstrap popover

### PR DESCRIPTION
**Pull request by ajainarayanan**
- The source of this issue is when we wanted to have a "bubble" sort of popovers in experiments view.
- Have a generic popover that looks close to what we have from design
- Have used popperjs to render the arrows based on the target
- Modifies `PlusButton` everywhere to use the same component (not in angular side)
  - Control Center
  - Entities detailed views
- Modifies more dropdown in dataprep to use popover

JIRA: https://issues.cask.co/browse/CDAP-12969

#### TODO:
- Need to convert all `Dropdown` and `UnControlledComponents/Popover` we use to `Popover`.

### CODE REVIEW

File: cdap-ui/app/cdap/components/Popover/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          <TargetElement />
+        </Target>
+        <Popper
+          placement={this.props.placement || 'auto'}
+          className={classnames("popper", {
+          'hide': !this.state.showPopover
+          })}
+        >
+          {this.props.children}
+          <Arrow className="popper__arrow"/>
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/Popover/index.js]: nit: space before `/>`

**Comment by tonybach** [comment]
Minor styling regressions:

1. Black background doesn't span full height of tooltip on fast actions:
 
screen shot 2017-12-21 at 4 35 54 pm

2. Additional space at top of workspace tab in DataPrep:

screen shot 2017-12-21 at 2 27 47 pm

**Comment by ajainarayanan** [comment]
@tonybach That is intentional. We are starting to have the plus button consistent across UI. Talked to @leacd about it before adding.

screen shot 2018-01-02 at 1 30 10 pm
screen shot 2018-01-02 at 1 29 57 pm
screen shot 2018-01-02 at 1 29 45 pm

**Comment by tonybach** [comment]
@ajainarayanan One more minor styling regression:

screen shot 2018-01-03 at 12 29 37 pm

The 'caret' tab (that shows when there are more tabs than available space) doesn't stick to the bottom border like the other tabs.

**Review by tonybach — APPROVED** [review]
LGTM 👍

**Comment by leacd** [comment]
LGTM

**Pull request merged**